src/timeline/js/controllers.js

```diff
@@ -671,83 +670,122 @@ App.controller('TimelineCtrl',function($scope) {
 
  // Select clip in scope
  $scope.SelectClip = function(clip_id, clear_selections, event) {
- 	// Trim clip_id
- 	var id = clip_id.replace("clip_", "");
+	// Trim clip_id
+	var id = clip_id.replace("clip_", "");
+
+	// Is CTRL pressed?
+	is_ctrl = false;
+	if (event && event.ctrlKey)
+		is_ctrl = true;
 
 	// Clear transitions also (if needed)
 	if (id != "" && clear_selections) {
-		$scope.SelectTransition("", clear_selections);
-		$scope.SelectEffect("", clear_selections);
+		if (is_ctrl)
+			$scope.SelectTransition("", false);
+		else
+			$scope.SelectTransition("", true);
 	}
+
 	// Call slice method and exit (don't actually select the clip)
 	if (id != "" && $scope.enable_razor) {
-        if ($scope.Qt) {
+		if ($scope.Qt) {
 			cursor_seconds = $scope.GetJavaScriptPosition(event.clientX);
-            timeline.RazorSliceAtCursor(id, "", cursor_seconds);
-        }
+			timeline.RazorSliceAtCursor(id, "", cursor_seconds);
+		}
 		// Don't actually select clip
-        return;
-    }
+		return;
+	}
+
 	// Is CTRL pressed?
 	is_ctrl = false;
-	if (event && event.ctrlKey) {
+	if (event && event.ctrlKey)
 		is_ctrl = true;
-	}
 
- 	// Unselect all clips
+	// Unselect all clips
 	for (var clip_index = 0; clip_index < $scope.project.clips.length; clip_index++) {
 		if ($scope.project.clips[clip_index].id == id) {
-			$scope.project.clips[clip_index].selected = true;
-			if ($scope.Qt) {
-				timeline.addSelection(id, "clip", clear_selections);
+			// Invert selection if CTRL is pressed and not forced add and already selected
+			if (is_ctrl && clear_selections && ($scope.project.clips[clip_index].selected == true)) {
+				$scope.project.clips[clip_index].selected = false;
+				if ($scope.Qt)
+					timeline.removeSelection($scope.project.clips[clip_index].id, "clip");
```

Comment:
I feel like `is_ctrl` can be hoisted up out of all of this, since ultimately when <kbd>Ctrl</kbd> is pressed, you don't want to do _anything_ to any of the clips except the one involved in the selection. So it might make the logic simpler to follow if we just get that out of the way first thing:
```js
for (var clip_index = 0; clip_index < $scope.project.clips.length; clip_index++) {
    if (is_ctrl && $scope.project.clips[clip_index].id !== id) {
      // Don't select/deselect any other clips, if Ctrl is pressed
      continue;
    }
```
That'll skip the entire rest of the loop body for every clip except the one we're interested in, and it hopefully makes the rest of the block a bit simpler too.

Reply:
Optimization? @ferdnyc , code even wasn't accepted yet, and you already making second improvement. Where you've been before?

Reply:
Right now I'm not seeing this as an improvement. <kbd>Ctrl</kbd> is rarely pressed. Simple click (or right click) is far more often operation in the Timeline over the items.